models/sipnet/R/read_restart.SIPNET.R

```diff
@@ -111,6 +117,8 @@ read_restart.SIPNET <- function(outdir, runid, stop.time, settings, var.names, p
   if ("TotSoilCarb" %in% var.names) {
     forecast[[length(forecast) + 1]] <- ens$TotSoilCarb[last]  ## kgC/m2
     names(forecast[[length(forecast)]]) <- c("TotSoilCarb")
+  }else{
+    params$restart["TotSoilCarb"] <- ens$TotSoilCarb[last]
   }
   
   print(runid)
```

Comment:
I think in addition to the read_restart function, you also need to make changes to the write_restart function, in the `write_restart_sipnet` function, the `params$restart` becomes `IC_extra` object, and you will need to write several `else` control just like what you did in the read_restart function to make sure that you pass the restart values from `IC_extra` to `analysis.save` object and name them accordingly.